src/App.js

```diff
@@ -64,8 +67,38 @@ export default class App extends Component {
     return this.state.languages.hasOwnProperty(lang);
   }
 
-  onRunParser(e) {
-    throw new Error('run parser not implemented yet!');
+  onRunParser() {
+    this.setState({ loading: true });
+    api.parse(this.currentLanguage, this.state.code)
+      .then(({ errors, status, uast }) => {
+        if (status === 'ok') {
+          this.setState({ ast: uast, loading: false });
+        } else {
+          this.setState({
+            loading: false,
+            errors: errors.map(err => {
+              if (typeof err === 'object' && err.hasOwnProperty('message')) {
+                return err.message;
+              } else if (typeof err === 'string') {
+                return err;
+              }
+
+              return null;
+            }).filter(err => err),
+          });
+
+          setTimeout(() => {
+            this.setState({ errors: [] });
+          }, 5000);
```

Comment:
Since we have now a `Notification` component I'd turn it a kind of service, so you could trigger an error, and then it would handle its rise and dismiss timeout